src/server/engine_shard.h

```diff
@@ -207,11 +207,18 @@ class EngineShard {
   void Shutdown();  // called before destructing EngineShard.
 
   void StartPeriodicFiber(util::ProactorBase* pb, std::function<void()> shard_handler);
+  void StartPeriodicFiberWithoutHeartbeat(util::ProactorBase* pb,
+                                          std::function<void()> shard_handler);
+  void StartPeriodicFiberImpl(util::ProactorBase* pb, std::function<void()> shard_handler,
+                              bool heartbeat);
 
   void Heartbeat();
   void RetireExpiredAndEvict();
 
-  void RunPeriodic(std::chrono::milliseconds period_ms, std::function<void()> shard_handler);
+  void RunHeartbeatPeriodic(std::chrono::milliseconds period_ms,
```

Comment:
why does heartbeat still gets shard handler?

Reply:
I wanted to keep the callback just in case we want to use it the future but I agree it redundant. I will remove